components/script/dom/gpudevice.rs

```diff
@@ -410,4 +412,70 @@ impl GPUDeviceMethods for GPUDevice {
 
         GPUBindGroupLayout::new(&self.global(), self.channel.clone(), bgl, binds, valid)
     }
+
+    /// https://gpuweb.github.io/gpuweb/#dom-gpudevice-createpipelinelayout
+    fn CreatePipelineLayout(
+        &self,
+        descriptor: &GPUPipelineLayoutDescriptor,
+    ) -> DomRoot<GPUPipelineLayout> {
+        // TODO: We should have these limits on device creation
+        let limits = GPULimits::empty();
+        let mut valid = true;
+        let mut bind_group_layouts = Vec::new();
+        let mut bgl_ids = Vec::new();
+        let mut max_dynamic_uniform_buffers_per_pipeline_layout =
+            limits.maxDynamicUniformBuffersPerPipelineLayout as i32;
+        let mut max_dynamic_storage_buffers_per_pipeline_layout =
+            limits.maxDynamicStorageBuffersPerPipelineLayout as i32;
+        descriptor.bindGroupLayouts.iter().for_each(|each| {
+            if each.is_valid() {
+                let id = each.id();
+                bind_group_layouts.push(id);
+                bgl_ids.push(id.0);
+            }
+            each.bindings().iter().for_each(|bind| {
+                match bind.type_ {
+                    GPUBindingType::Uniform_buffer => {
+                        if bind.hasDynamicOffset {
+                            max_dynamic_uniform_buffers_per_pipeline_layout -= 1;
+                        };
+                    },
+                    GPUBindingType::Storage_buffer => {
+                        if bind.hasDynamicOffset {
+                            max_dynamic_storage_buffers_per_pipeline_layout -= 1;
+                        };
+                    },
+                    GPUBindingType::Readonly_storage_buffer => {
+                        if bind.hasDynamicOffset {
+                            max_dynamic_storage_buffers_per_pipeline_layout -= 1;
+                        };
+                    },
+                    _ => {},
+                };
+            });
+        });
+
+        valid &= descriptor.bindGroupLayouts.len() <= limits.maxBindGroups as usize &&
```

Comment:
It looks like we can declare `let valid =` here, rather than declaring it as mutable earlier.